### PR TITLE
fixed a few minor bugs and support pytorch0.3

### PR DESCRIPTION
**Pull request by yuanzhaoYZ**
1. convert feed_interpolated_input input x to x.cuda() if cuda is available
2. added `dim` parameter for torch.mean() so that the code base can work with pytorch 0.3

### CODE REVIEW

**Comment by nashory** [comment]
great work! :)
